test/e2e/tests/test_references.py

```diff
@@ -0,0 +1,315 @@
+    ref = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, 'dbclusterparametergroups',
+        resource_name, namespace="default",
+    )
+    k8s.create_custom_resource(ref, resource_data)
+    cr = k8s.wait_resource_consumed_by_controller(ref)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(ref)
+
+    yield ref, cr, resource_name
+
+    # Try to delete, if doesn't already exist
+    try:
+        _, deleted = k8s.delete_custom_resource(ref, 3, 10)
+        assert deleted
+        time.sleep(DELETE_WAIT_AFTER_SECONDS)
+    except:
+        pass
+
+    db_cluster_parameter_group.wait_until_deleted(resource_name)
+
+
+@pytest.fixture(scope="module")
+def ref_db_cluster(k8s_secret, db_cluster_name, cpg_name):
+    db_name = "mydb"
+    secret = k8s_secret(
+        MUP_NS,
+        random_suffix_name(MUP_SEC_CLUSTER_NAME_PREFIX, 32),
+        MUP_SEC_KEY,
+        MUP_SEC_VAL,
+    )
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["DB_CLUSTER_ID"] = db_cluster_name
+    replacements["DB_NAME"] = db_name
+    replacements["MASTER_USER_PASS_SECRET_NAMESPACE"] = secret.ns
+    replacements["MASTER_USER_PASS_SECRET_NAME"] = secret.name
+    replacements["MASTER_USER_PASS_SECRET_KEY"] = secret.key
+    replacements["DB_CLUSTER_PARAMETER_GROUP_NAME"] = cpg_name
+
+    resource_data = load_rds_resource(
+        "db_cluster_ref",
+        additional_replacements=replacements,
+    )
+
+    ref = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, 'dbclusters',
+        db_cluster_name, namespace="default",
+    )
+    k8s.create_custom_resource(ref, resource_data)
+    cr = k8s.wait_resource_consumed_by_controller(ref)
+
+    # NOTE(jaypipes): We specifically do NOT wait for the DBInstance to exist
+    # in the RDS API here because we will create the referred-to
+    # DBClusterParameterGroup and wait for the reference to be resolved
+
+    yield (ref, cr, db_cluster_name)
+
+    if k8s.get_resource_exists(ref):
+        # If all goes properly, we should not hit this because the test cleans
+        # up the child resource before exiting...
+        _, deleted = k8s.delete_custom_resource(
+            ref,
+            period_length=DELETE_INSTANCE_TIMEOUT_SECONDS,
+        )
+        assert deleted
+
+        db_cluster.wait_until_deleted(db_cluster_name)
+
+
+@pytest.fixture
+def ref_db_instance(k8s_secret, db_cluster_name, pg_name):
+    db_instance_id = random_suffix_name("pg14-t3-micro", 20)
+    secret = k8s_secret(
+        MUP_NS,
+        random_suffix_name(MUP_SEC_INSTANCE_NAME_PREFIX, 32),
+        MUP_SEC_KEY,
+        MUP_SEC_VAL,
+    )
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements['COPY_TAGS_TO_SNAPSHOT'] = "False"
+    replacements["DB_INSTANCE_ID"] = db_instance_id
+    replacements["DB_CLUSTER_ID"] = db_cluster_name
+    replacements["DB_PARAMETER_GROUP_NAME"] = pg_name
+    replacements["MASTER_USER_PASS_SECRET_NAMESPACE"] = secret.ns
```

Comment:
I don't think these `MASTER_USER_PASS_SECRET...` template vars are used for a DB instance resource

Reply:
@jljaco copy pasta :) removed the unused variables.